Rotated hexagons

mcnees opened the pull request:
Rewrote the hexagon and triangle pattern functions to include a rotation parameter. Removed the iso pattern code, changed the iso option to call triangles[rotate=90], added a hexup option that calls hexagons[rotate=90].

duetosymmetry commented on gridpapers.sty:
```diff
@@ -195,26 +195,33 @@
+    % tile size/.store in=\tikztilesize,tile size=10pt,
+	}, 
+  code={
+	 	\pgfsetlinewidth{0.6pt}
+	    \pgftransformshift{\pgfpoint{0mm}{0.866025*\GP@patternsize}}
+	    \pgfpathmoveto{\pgfpoint{0mm}{0mm}}
+	    \pgfpathlineto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{-0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{-0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{3*\GP@patternsize}{0mm}}
+	    \pgfpathmoveto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{0.866025*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	    \pgfusepath{stroke}
```
Mixing tabs and spaces, can we try to avoid tabs?

mcnees replied:
Yes!

duetosymmetry commented on gridpapers.sty:
```diff
@@ -195,26 +195,33 @@
+  tile transformation={rotate=\tikzhexrotate},
+  defaults={
+    rotate/.store in=\tikzhexrotate,rotate=0,
+    % tile size/.store in=\tikztilesize,tile size=10pt,
```
Do we need to keep this commented line around?

mcnees replied:
No!

duetosymmetry commented on graph paper.tex:
```diff
@@ -1,8 +1,8 @@
-~
-\end{document}
+~ 
+\end{document} 
```
Does this file need to change for this commit?

mcnees replied:
No!

duetosymmetry commented on .graph paper.watcher_pid:
```diff
@@ -0,0 +1 @@
+89775
```
Please remove this file, and the files `.graph paper.lb`, `.DS_Store` from the commit

mcnees replied:
Will do

duetosymmetry commented on graph paper.tex:
```diff
@@ -1,6 +1,6 @@
+\usepackage[pattern=iso,minorcolor=cornflower!30]{gridpapers}
 
 \begin{document}  
 \thispagestyle{empty}
```
Does this file need to be touched for this commit?

mcnees replied:
No, I’m being super sloppy

duetosymmetry commented on gridpapers.sty:
```diff
@@ -232,53 +232,46 @@
+    % tile size/.store in=\tikztilesize,tile size=10pt,
+	}, 
+  code={
+	 	% \pgfsetlinewidth{0.6pt}
+	 	% 	    \pgftransformshift{\pgfpoint{0mm}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathmoveto{\pgfpoint{0mm}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{-0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{-0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{3*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathmoveto{\pgfpoint{0.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfusepath{stroke}
```
Pretty sure we can delete these commented lines

mcnees replied:
Yes, I forgot to save the file in my editor before pushing the changes

duetosymmetry commented on gridpapers.sty:
```diff
@@ -232,53 +232,46 @@
+	 	% 	    \pgfpathlineto{\pgfpoint{2*\GP@patternsize}{0.866025*\GP@patternsize}}
+	 	% 	    \pgfpathlineto{\pgfpoint{2.5*\GP@patternsize}{0mm}}
+	 	% 	    \pgfusepath{stroke}
+	      \pgfsetlinewidth{0.6pt}
+	    \pgfpathmoveto{\pgfpoint{0mm}{0mm}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{2*0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{0mm}{2*0.8660254*\GP@patternsize}}
+	      \pgfpathmoveto{\pgfpoint{0mm}{0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0.8660254*\GP@patternsize}}
+	      \pgfpathmoveto{\pgfpoint{0mm}{2*0.8660254*\GP@patternsize}}
+	    \pgfpathlineto{\pgfpoint{\GP@patternsize}{0mm}}
+	    \pgfpathlineto{\pgfpoint{0mm}{0mm}}
+	      \pgfusepath{stroke}
```
Mixing tabs and spaces, let's try to avoid tabs

mcnees replied:
Got it

duetosymmetry reviewed (changes requested):
Looking nice! Besides the comments below, we should also:
- Update the README with the new pattern
- Update the `.dtx` with both the new code, and documentation of new pattern
- Bump the version number and add a `\changes{}` entry in the `.dtx`
These can be done in a separate documentation commit.

Do you know how to amend/rebase your commits?

mcnees commented:
Follow-up: I do not know how to amend/rebase my commits.

duetosymmetry commented:
Ok, go ahead and add more commits onto the end of the history here. When we merge we will use the 'squash and merge' option and it will squash everything down to one commit. If you undo the unwanted changes, they will get squashed down to nothing.

mcnees commented:
Okay, I think I got it all.

duetosymmetry commented:
Almost there... note that if you use the `.ins` and `.dtx` files to generate the `.sty` file, the generated file does not match what you have in the commit. That is, try doing the following:

    pdflatex gridpapers.ins
(and say 'y' when it prompts you to overwrite). Then with a git diff you will see that the generated `.sty` file differs.

Other small changes: the .DS_Store file is still in the commit, and there is a diff between master and this branch in the file 'graph paper.tex'. That latter diff is that there used to be a newline terminating the file, but the newline was deleted.

mcnees commented:
Okay, the .sty file in the commit is the one generated from the .ins and .dtx files, and I think everything else has been fixed?

duetosymmetry commented:
Looks almost ready to me! The last thing is that when we compile the documentation from the .dtx file, we get a warning about the checksum not passing,

    ! Package doc Error: Checksum not passed (432<>405).

Their simplistic approach to a checksum is just counting number of backslashes; we have to update this number whenever we update the .dtx. Just change [the 432 on line 53](https://github.com/mcnees/LaTeX-Graph-Paper/blob/68762a7e3dad067a23aa9d66f0e1e813632fbff3/gridpapers.dtx#L53) to a 405.

If you want to feel really virtuous, there is also trailing whitespace ending lines 76, 126, 188, 378, and 678; this gets stripped when generating the .sty.

mcnees commented:
Fixed the checksum and trailing whitespace, and also replaced

    [2021/03/27 v1.0.2 Graph paper backgrounds]

after \ProvidesPackage{gridpapers}.

You can probably tell I've never used github for anything more than stashing files.

mcnees commented:
Whoops, that changes the line in the .sty file, so I had to update that too. Jfc.

duetosymmetry commented:
> Fixed the checksum and trailing whitespace, and also replaced
> 
> ```
> [2021/03/27 v1.0.2 Graph paper backgrounds]
> ```
> 
> after \ProvidesPackage{gridpapers}.
> 

Ah, good catch! I missed that. LGTM, going to hit squash and merge! Yippee!
